br/pkg/utils/misc.go

```diff
@@ -239,6 +241,17 @@ func FlattenValues[K comparable, V any](m map[K][]V) []V {
 	return result
 }
 
+// GetPartitionByName gets the partition ID from the given tableInfo if its name matches
+func GetPartitionByName(tableInfo *model.TableInfo, name ast.CIStr) (int64, error) {
+	if tableInfo.Partition == nil {
+		return 0, errors.Errorf("the table %s[id=%d] does not have parition", tableInfo.Name.O, tableInfo.ID)
+	}
+	if partID := tableInfo.Partition.GetPartitionIDByName(name.L); partID > 0 {
```

Comment:
could this name be empty string? Jus asking

Reply:
No